### ISSUE TITLE
Failed to compile 1.1.0

### ISSUE THREAD

**Issue by mo8it** [01mf02/jaq#120]
I get this error when trying to update the binary with `cargo-update`:

```
   Compiling jaq v1.1.0
error[E0282]: type annotations needed
   --> /var/home/mo8it/.cargo/registry/src/index.crates.io-6f17d22bba15001f/jaq-1.1.0/src/main.rs:441:57
    |
441 |                 ColoredFormatter::new(CompactFormatter).write_colored_json(
    |                                                         ^^^^^^^^^^^^^^^^^^ cannot infer type of the type parameter `T` declared on the method `write_colored_json`
    |
help: consider specifying the generic arguments
    |
441 |                 ColoredFormatter::new(CompactFormatter).write_colored_json::<_, T>(
    |                                                                           ++++++++

For more information about this error, try `rustc --explain E0282`.
error: could not compile `jaq` (bin "jaq") due to previous error
error: failed to compile `jaq v1.1.0`, intermediate artifacts can be found at `/tmp/cargo-installnQRe5x`.
```

**Comment by kklingenberg** [comment]
Could you include the cargo and rustc versions you're using?

```bash
cargo --version
rustc --version
```

**Comment by mo8it** [comment]
I am using the latest stable Rust version on Linux.

`cargo --version`:

```
cargo 1.73.0
```

`rustc --version`:

```
rustc 1.73.0
```

The problem is probably because `cargo-update` doesn't use `--locked` by default. I know that the official installation instructions for `jaq` recommend using `--locked`, but many prefer installation without it to receive bug fixes.

I think that the problem will arise anyway after updating the dependencies.
Therefore, I would suggest updating the dependencies, fixing that small compiler error and publishing `1.1.2` ❤️

**Comment by ctron** [comment]
I re-released the version 3.3.0 as 4.0.0 and also yanked the 3.3.0 release. Sorry for the troubles.

**Comment by mo8it** [comment]
I can confirm that compilation works now after the version 3.3.0 is yanked. @ctron Thanks 🥰
